pkg/virt-api/rest/subresource.go

```diff
@@ -1545,3 +1555,261 @@ func (app *SubresourceAPIApp) RemoveMemoryDumpVMRequestHandler(request *restful.
 
 	response.WriteHeader(http.StatusAccepted)
 }
+
+func generateVMIInterfaceRequestPatch(vmi *v1.VirtualMachineInstance, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if alreadyExists(interfaceRequest, iface) {
+			return "", fmt.Errorf("unable to add interface [%s] because it already exists", iface.Name)
+		}
+	}
+
+	vmiCopy := vmi.DeepCopy()
+	vmiCopy.Spec = *ApplyInterfaceRequestOnVMISpec(&vmiCopy.Spec, interfaceRequest)
+
+	oldIfacesJSON, err := json.Marshal(vmi.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return "", err
+	}
+
+	newIfacesJSON, err := json.Marshal(vmiCopy.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return "", err
+	}
+
+	oldNetworksJSON, err := json.Marshal(vmi.Spec.Networks)
+	if err != nil {
+		return "", err
+	}
+
+	newNetworksJSON, err := json.Marshal(vmiCopy.Spec.Networks)
+	if err != nil {
+		return "", err
+	}
+
+	testNetworks := fmt.Sprintf(`{ "op": "test", "path": "/spec/networks", "value": %s}`, string(oldNetworksJSON))
+	updateNetworks := fmt.Sprintf(`{ "op": "%s", "path": "/spec/networks", "value": %s}`, "replace", string(newNetworksJSON))
+
+	testInterfaces := fmt.Sprintf(`{ "op": "test", "path": "/spec/domain/devices/interfaces", "value": %s}`, string(oldIfacesJSON))
+	updateInterfaces := fmt.Sprintf(`{ "op": "%s", "path": "/spec/domain/devices/interfaces", "value": %s}`, "replace", string(newIfacesJSON))
+
+	patch := fmt.Sprintf("[%s, %s, %s, %s]", testNetworks, testInterfaces, updateNetworks, updateInterfaces)
```

Comment:
This is interesting.

First some tooling comment: Consider using https://github.com/kubevirt/kubevirt/blob/main/pkg/apimachinery/patch/patch.go .

But the fun part is...
Lets try and investigate if this can be simplified.
For an addition, the patch can just add a specific interface to the list ([see here](https://jsonpatch.com/#add)).

Another point is the testing part. Why would we be interested to check that no one else collided when updating the list?
If we replace the whole list, we indeed need to "lock" it by using the test, but if we add it individually, we do not need to.

It will be a different story for the unplug scenario.